### ISSUE TITLE
DeOldify in web browser using WebAssembly

### ISSUE THREAD

**Issue by niutech** [jantic/DeOldify#397]
What would it take to run DeOldify with a pretrained model in a web browser using Python in WebAssembly, like [Pyodide](https://pyodide.org)? For simplicity, it could only colorize single images. Would it be possible, provided that WebAssembly provides [SIMD](https://v8.dev/features/simd) instructions and [WebGPU](https://web.dev/gpu/) is available in browsers under a flag? See also [Compiling ML to WASM and WebGPU with Apache TVM](https://tvm.apache.org/2020/05/14/compiling-machine-learning-to-webassembly-and-webgpu).

**Comment by jantic** [comment]
@niutech - Research, persistence, and yet knowing when to quit! Most likely it won't work the first time as it seems almost everything in the current ecosystem breaks when it comes to interoperability between formats for anything that's outside of "normal"- that is, basic vision networks like resnet and nlp like BERT.  I can already tell you two candidates for likely headaches: The hooks generated by hook_outputs in unet.py, and spectral norm.  The former could potentially be remedied by [tracing](https://pytorch.org/docs/stable/generated/torch.jit.trace.html) the model with the jit and using the resulting torchscript model.  The latter is easily remedied by simply stripping spectral norm off the generator model after loading it and using that instead.  You can use the function I posted [here](https://github.com/jantic/DeOldify/issues/166#issuecomment-540882832) . 

Now more headaches to consider:  Realistically you may have to upgrade to the latest PyTorch (1.10.0) in order to get torchscript compiles working well (it's your best shot anyway).  Yet if you do that you're going to have to fork a custom fastai 1.0.51 and get it working with PyTorch 1.10.0 (we're moving from PyTorch 1.0.1 here).  Another user described that process [here](https://github.com/jantic/DeOldify/issues/354).

tl;dr :  It's non trivial and will require research, persistence, and some luck.
